### PR TITLE
lower minimal mbr gap size to allow creating unaligned partitions

### PR DESCRIPTION
**Pull request by wfeldt**
For ms-dos partition tables the default minimal_mbr_gap setting of 1 MiB
prevents creating partitions in the expert partitioner starting at a lower
value (via the region dialog).

Note this is not an issue with gpt, there you can already do this.

### CODE REVIEW

**Comment by ancorgs** [comment]
Well, if allowing to create misaligned partitions is important for the Partitioner. This looks like a safe place to put that code, although I miss a explanatory comment.

**Comment by ancorgs** [comment]
And I guess we are sure than introducing this change only opens the door to specify misaligned partitions using the "custom region" option. "Maximum size" and "custom size" should keep producing aligned partitions, no matter what.

File: src/lib/y2partitioner/actions/create_partition_table.rb [hunk 1]
```diff
@@ -77,7 +77,10 @@ def perform_action
       # @param type [Y2Storage::PartitionTables::Type]
       def create_partition_table(type)
         disk.remove_descendants
-        disk.create_partition_table(type)
+        ptable = disk.create_partition_table(type)
+        if ptable.type.is?(:msdos)
+          ptable.minimal_mbr_gap = Y2Storage::DiskSize.B(512)
```
Comment (joseivanlopez) [inline src/lib/y2partitioner/actions/create_partition_table.rb]: After talking with Ancor now I understood why we want to create a non optimal aligned mbr gap, but, why 512 B?

Comment (joseivanlopez) [reply src/lib/y2partitioner/actions/create_partition_table.rb]: When we are creating a partition without custom region option, does the resulting partition continue being optimal aligned at the beginning?

Comment (wfeldt) [reply src/lib/y2partitioner/actions/create_partition_table.rb]: 0.5 KiB is the space used by the partition table itself. You cannot go below that.

As already said, this only brings us on par with gpt where this is already possible. This does not affect partition alignment in any way.

Comment (joseivanlopez) [reply src/lib/y2partitioner/actions/create_partition_table.rb]: I would appreciate if you append this explanation to the comment in the code.

Comment (ancorgs) [reply src/lib/y2partitioner/actions/create_partition_table.rb]: Our documentation of `PartitionTables::Msdos#mbr_gap` is not clear/precise enough, which makes those 512 B confusing. See https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/partition_tables/msdos.rb#L44

That states that mbr_gap is "_the space between the **end** of the MBR and the beginning of the first partition._". That is not true. Is the space between the START of the MBR (ie. the start of the disk) and the beginning of the first partition. Isn't it?

Comment (ancorgs) [reply src/lib/y2partitioner/actions/create_partition_table.rb]: Note that documentation is only at `Y2Storage` level. There is no such method in libstorage-ng. In the library there is only `minimal_mbr_gap` (which is not documented).

Comment (wfeldt) [reply src/lib/y2partitioner/actions/create_partition_table.rb]: > Is the space between the START of the MBR (ie. the start of the disk) and the beginning of the first partition. Isn't it?

Yes.

**Pull request merged**